gcp_variant_transforms/libs/processed_variant_test.py

```diff
@@ -13,15 +13,41 @@
 # limitations under the License.
 
 from __future__ import absolute_import
+from typing import Dict  # pylint: disable=unused-import
 
 import unittest
 
 from vcf import parser
 
 from gcp_variant_transforms.beam_io import vcfio
+from gcp_variant_transforms.libs import metrics_util
 from gcp_variant_transforms.libs import processed_variant
 from gcp_variant_transforms.libs import vcf_header_parser
 
+class CounterMock(metrics_util.CounterInterface):
```

Comment:
Following up on a comment that I left to be resolved after design agreement: Since we agreed to keep injecting the factory, I think this solution is superior over `patch`. The name `Mock` is really poorly chosen for these, so I renamed them to Spies following this [nomenclature](https://martinfowler.com/articles/mocksArentStubs.html#TheDifferenceBetweenMocksAndStubs). Also I chose to keep them here as module privates for now (as the only user is this module). Once we have a second usage for them we should move them to the `testing` directory.